### PR TITLE
adding CreateVolume function to docker_client

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
CreateVolume function added to docker_client

### Implementation details
A wrapper for docker.CreateVolume API is added to docker_client.go
For the interest of keeping commits small, I'll add remaining volume APIs (list, inspect, and delete) separately.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: yes (unit tests added)

### Description for the changelog
changelog not updated

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/docker_client_test.go [hunk 1]
```diff
@@ -1282,3 +1282,67 @@ func TestMetadataFromContainerHealthCheckWithNoLogs(t *testing.T) {
...
+	volumeName := "volumeName"
+	mountPoint := "some/mount/point"
+	driver := "driver"
+	driverOptions :=  map[string]string{
+		"opt1": "val1",
+		"opt2": "val2",
+	}
+	gomock.InOrder(
+		mockDocker.EXPECT().CreateVolume(gomock.Any()).Do(func(opts docker.CreateVolumeOptions) {
+			if opts.Name != volumeName {
```
Comment (aaithal) [inline agent/engine/docker_client_test.go]: you can use `assert.Equals()` for all of these comparisons.

File: agent/engine/docker_client_test.go [hunk 2]
```diff
@@ -1282,3 +1282,67 @@ func TestMetadataFromContainerHealthCheckWithNoLogs(t *testing.T) {
 	metadata := metadataFromContainer(dockerContainer)
 	assert.Equal(t, api.ContainerUnhealthy, metadata.Health.Status)
 }
+
+func TestCreateVolumeTimeout(t *testing.T) {
+	timeout := 1*time.Second
```
Comment (aaithal) [inline agent/engine/docker_client_test.go]: is the test actually waiting for this duration? If yes, can this be reduced to something in the order of milliseconds (Example: `10 ms`)?

File: agent/engine/docker_client.go [hunk 3]
```diff
@@ -962,6 +966,78 @@ func (dg *dockerGoClient) Version() (string, error) {
...
+type volumeResponse struct {
+	Volume *taskresource.VolumeResource
+	Error       error
+}
+
+func (dg *dockerGoClient) CreateVolume(name string,
+	mountpoint string,
+	driver string,
+	driverOptions map[string]string,
+	labels map[string]string, timeout time.Duration) volumeResponse {
```
Comment (aaithal) [inline agent/engine/docker_client.go]: nit: `timeout` needs to be in a newline i think

**Comment by yunhee-l** [comment]
I will add the integ tests when I rest of the APIs implemented, so that I can use listVolumes or inspectVolume, so that I can have a way to verify that the volume was created, and removeVolume after the test.

**Pull request merged**